### PR TITLE
Calculate hash of system message attributes

### PR DESCRIPTION
**Pull request by katlasik**

### CODE REVIEW

File: rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala [hunk 1]
```diff
@@ -30,66 +29,30 @@ trait SendMessageDirectives {
...
-          complete(SendMessageResponse(messageAttributeDigest, digest, None, message.id.id, message.sequenceNumber))
+        doSendMessage(queueActor, message).map {
+          case MessageSendingResult(message, digest, messageAttributeDigest, messageSystemAttributeDigest) =>
+            complete(SendMessageResponse(messageAttributeDigest, digest, messageSystemAttributeDigest, message.id.id, message.sequenceNumber))
         }
       }
     }
   }
 
-  def getMessageAttributes(parameters: Map[String, String]): Map[String, MessageAttribute] = {
```
Comment (katlasik) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala:40]: Two functions were doing basically same job with parsing arguments -> one for message system attributes and other for attributes. I remove one and generalized other.

File: rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala [hunk 2]
```diff
@@ -252,6 +222,8 @@ trait SendMessageDirectives {
     }
   }
 
+  case class MessageSendingResult(data: MessageData, digest: String, messageAttributeDigest: Option[String], systemMessageAttributeDigest: Option[String])
```
Comment (adamw) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala:225]: wouldn't `MessageSendResult` bo more "english"?

File: rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala [hunk 3]
```diff
@@ -30,66 +29,30 @@ trait SendMessageDirectives {
...
-      case (messageSystemAttributeNamePattern(index), parameterName) =>
-        val parameterDataType = parameters(s"MessageSystemAttribute.$index.Value.DataType")
+      case (messageAttributeNamePattern(index), parameterName) =>
+        val parameterDataType = parameters(s"$prefix.$index.Value.DataType")
+
         val parameterValue = parameterDataType match {
-          case "String" => StringMessageAttribute(parameters(s"MessageSystemAttribute.$index.Value.StringValue"))
-          case "Number" => NumberMessageAttribute(parameters(s"MessageSystemAttribute.$index.Value.StringValue"))
+          case "String" => StringMessageAttribute(parameters(s"$prefix.$index.Value.StringValue"))
+          case "Number" => NumberMessageAttribute(parameters(s"$prefix.$index.Value.StringValue"))
```
Comment (adamw) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala:49]: I know it's been like that in the old impl, but shouldn't this be different than `StringValue`?

**Review by adamw — APPROVED** [review]

File: rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala (hunk 3 above)
Comment (katlasik) [reply rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala:49]: No:

> Amazon SQS supports the following logical data types: String, Number, and Binary. For the Number data type, you must use StringValue.

File: rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala (hunk 2 above)
Comment (katlasik) [reply rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala]: fixed

File: rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala (hunk 3 above)
Comment (adamw) [reply rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala:49]: Interesting design choice ;)

**Comment by micossow** [comment]
closes #827

**Pull request merged**